### PR TITLE
Cgi execve parsing

### PR DESCRIPTION
**Pull request by rosamakinen**
added parsing the arguments for execve, checking for executing rights instead of file existing, added couple more environment variables.

basically the rough core of it works now, next needing to add the query_string and request_uri variables to be able to hopefully run the hello_world.py

### CODE REVIEW

File: include/WebServer.hpp [hunk 1]
```diff
@@ -8,6 +8,7 @@
 #define HTTP_LINEBREAK		"\r\n"
 #define SPACE				" "
 #define WHITESPACE			" \t"
+#define PREFIX_SHEBANG		"#!"
```
Comment (merituulie) [inline include/WebServer.hpp:11]: I mean personally I love it but can we get something more descriptive name for this :D

**Review by forbidden-arts — APPROVED** [review]

File: src/CgiHandler.cpp [hunk 2]
```diff
@@ -38,14 +35,52 @@ static char **transferToString(std::map<std::string, std::string> cgiEnvironment
...
 
-int CgiHandler::executeCgi(HttpRequest request)
+std::string findShebang(std::string fullPath)
 {
+	std::ifstream file(fullPath);
+	std::string	line;
 	
-	std::map<std::string, std::string> cgiEnvironment = initCgiEnvironment(request);
-	char **variableArray = transferToString(cgiEnvironment);
+	if (file.is_open() == true)
```
Comment (merituulie) [inline src/CgiHandler.cpp:43]: we should follow the filehandlers way of doing this like if the file is not open or is bad etc we throw right?

File: include/WebServer.hpp (hunk 1 above)
Comment (rosamakinen) [reply include/WebServer.hpp:11]: ahaha sure, I could not think of anything more descreptive, so I'll take ideas :D

File: src/CgiHandler.cpp (hunk 2 above)
Comment (rosamakinen) [reply src/CgiHandler.cpp:43]: sure, I'll fix it for the next one!

File: include/WebServer.hpp (hunk 1 above)
Comment (merituulie) [reply include/WebServer.hpp:11]: Is this script prefix or smt? You can push it in the next thing too if you're working on the same thing still after this PR!

**Review by merituulie — APPROVED** [review]

**Pull request merged**